client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -165,4 +184,24 @@ class PaymentMethodStripe extends Component {
 	}
 }
 
-export default localize( PaymentMethodStripe );
+function mapStateToProps( state ) {
+	const site = getSelectedSiteWithFallback( state );
+	const isRequesting = getIsRequesting( state, site.ID );
+	const stripeConnectAccount = getStripeConnectAccount( state, site.ID );
+	return {
+		isRequesting,
+		site,
```

Comment:
It looks like it  would make sense to destruct site to ID( as siteID ) and domain here. It would allow getting rid of a lot of checks and comparisons through the component. For example in `componentWillReceiveProps` you could then just do 
`if( this.props.siteID !== nextProps.siteID )`

Reply:
+1

Reply:
Excellent idea - i noticed a lot of rendering in various components (not just in this PR) due to site changing - this should help - https://github.com/Automattic/wp-calypso/pull/19115/commits/b8571c6a1613a1103339fd346e3b71afa617f33a